packages/monosize-storage-azure/src/uploadReportToRemote.mts

```diff
@@ -14,71 +16,112 @@ export function splitArrayToChunks<T>(arr: T[], size: number): T[][] {
+
+      if (shouldEntryBeDeleted) {
+        transaction.deleteEntity(entity.partitionKey as string, entity.rowKey as string);
+      }
     }
-  }
 
-  localReport.forEach(entry => {
-    transaction.upsertEntity(
-      {
-        partitionKey: branch,
-        rowKey: createRowKey(entry),
+    localReport.forEach(entry => {
+      transaction.upsertEntity(
+        {
+          partitionKey: branch,
+          rowKey: createRowKey(entry),
 
-        name: entry.name,
-        packageName: entry.packageName,
-        path: entry.path,
+          name: entry.name,
+          packageName: entry.packageName,
+          path: entry.path,
 
-        minifiedSize: entry.minifiedSize,
-        gzippedSize: entry.gzippedSize,
+          minifiedSize: entry.minifiedSize,
+          gzippedSize: entry.gzippedSize,
 
-        commitSHA,
-      },
-      'Replace',
+          commitSHA,
+        },
+        'Replace',
+      );
+    });
+
+    const chunks = splitArrayToChunks(transaction.actions, ENTRIES_PER_CHUNK);
+
+    for (const chunk of chunks) {
+      await client.submitTransaction(chunk);
+    }
+  }
+
+  return uploadReportToRemote;
+}
+
+function getTableClient(authType: AzureAuthenticationType): TableClient {
+  if (typeof process.env['BUNDLESIZE_ACCOUNT_NAME'] !== 'string') {
+    throw new Error('monosize-storage-azure: "BUNDLESIZE_ACCOUNT_NAME" is not defined in your process.env');
+  }
+  const AZURE_STORAGE_ACCOUNT = process.env['BUNDLESIZE_ACCOUNT_NAME'];
+  const AZURE_STORAGE_TABLE_NAME = 'latest';
+
+  if (authType === 'AzurePipelinesCredential') {
+    if (typeof process.env['AZURE_TENANT_ID'] !== 'string') {
+      throw new Error('monosize-storage-azure: "AZURE_TENANT_ID" is not defined in your process.env');
+    }
+    if (typeof process.env['AZURE_CLIENT_ID'] !== 'string') {
+      throw new Error('monosize-storage-azure: "AZURE_CLIENT_ID" is not defined in your process.env');
+    }
+    if (typeof process.env['AZURE_SERVICE_CONNECTION_ID'] !== 'string') {
+      throw new Error('monosize-storage-azure: "AZURE_SERVICE_CONNECTION_ID" is not defined in your process.env');
+    }
+    if (typeof process.env['SYSTEM_ACCESSTOKEN'] !== 'string') {
+      throw new Error('monosize-storage-azure: "SYSTEM_ACCESSTOKEN" is not defined in your process.env');
+    }
+    const tenantId = process.env['AZURE_TENANT_ID'];
+    const clientId = process.env['AZURE_CLIENT_ID'];
+    const serviceConnectionId = process.env['AZURE_SERVICE_CONNECTION_ID'];
+    const systemAccessToken = process.env['SYSTEM_ACCESSTOKEN'];
+
+    return new TableClient(
+      `https://${AZURE_STORAGE_ACCOUNT}.table.core.windows.net`,
+      AZURE_STORAGE_TABLE_NAME,
+      new AzurePipelinesCredential(tenantId, clientId, serviceConnectionId, systemAccessToken),
     );
-  });
+  } else {
+    /**
+     * Defaults to AzureNamedKeyCredential
+     */
+    if (typeof process.env['BUNDLESIZE_ACCOUNT_KEY'] !== 'string') {
```

Comment:
suggestion: a good practice is also to cover default explicitly and throw in all other cases ( is consumer is not using TS a typo can happen very easily )


```
  // 👉 handling default ( if not specified in user config )
  if (authType === 'AzureNamedKeyCredential') {
   // your logic ....

    return new TableClient();
  }

 // 👉 handling invalid auth type at the end
  throw new Error(`monosize-storage-azure: "authType: ${authType}" is not supported`);
```